polyply/src/generate_templates.py

```diff
@@ -302,9 +299,12 @@ def __init__(self, topology, max_opt, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.templates = {}
         self.resnames = []
-        self.volumes = {}
         self.max_opt = max_opt
         self.topology = topology
+        if "volumes" in self.topology:
+            self.volumes = self.topology.volumes
+        else:
+            self.volumes = {}
```

Comment:
Is topology a dict? If so:
```suggestion
        self.volumes = self.topology.get("volumes", {})
```